### ISSUE TITLE
you may not specify inputs or outputs when running a databricks job on an existing cluster

### ISSUE THREAD

**Issue by gison93** [Azure/MachineLearningNotebooks#240]
When building a pipeline with a DatabricksStep passing inputs, outputs and existing_cluster_id I get the following error in the shareableexecutionlogs "Failed to submit job because of error: System.ArgumentException: you may not specify inputs or outputs when running a databricks job on an existing cluster."

**Comment by mcb0035** [comment]
Is this expected?  Not being able to use parameters is a pretty big limitation.

**Comment by j-martens** [comment]
@parasharshah  @nacharya1 any information for the user?

**Comment by juergstaub** [comment]
Same here

**Comment by j-martens** [comment]
@sanpil @yanrez @parasharshah

**Comment by j-martens** [comment]
@juergstaub / @mcb0035 / @gison93 

Engineering shared this: The only i/o supported for existing clusters are DBFS (https://docs.microsoft.com/en-us/python/api/azureml-core/azureml.data.dbfs_datastore.dbfsdatastore?view=azure-ml-py)
azureml.data.dbfs_datastore.DBFSDatastore class - Azure Machine Learning Python
A Databricks File System (DBFS) Datastore. docs.microsoft.com​

For other I/O types there is a requirement to configure the cluster to access them which requires a restart of the cluster and we do not want to restart an existing cluster since jobs might be running on them. 

cc: @fahdkmsft

**Comment by j-martens** [comment]
Hopefully this answers your questions. I will proceed to now close this issue. Please reopen if this does not sufficiently clarify things. Thanks!

**Comment by mcb0035** [comment]
Is it possible to configure a cluster for I/O so it does not have to be
restarted, then use that cluster instead of creating a new one every time?
This would bypass the issue.

**Comment by j-martens** [comment]
@mcb0035 - this is indeed possible. Clusters can be pre-configured by the customer to access an I/O path. They have to ensure that they don't pass those paths as inputs or outputs to the DatabricksStep.

**Comment by gitsathish** [comment]
Could you please share pointers to a doc or steps to pre-configure a cluster to an I/O path ?

**Comment by sanpil** [comment]
Does this help? https://aka.ms/pl-databricks.

**Comment by shpraka** [comment]
@j-martens @sanpil By pre-configure you mean at the time creating the cluster right? What if I have created the cluster outside the azureml notebook and then want to add reference to adls store like below, is there any workaround to achieve that? 

````
adls_datastore = Datastore.get(ws, 'adls')
print(adls_datastore)

adls_input = DataReference(datastore=adls_datastore, path_on_datastore="SampleCDM",
                                   data_reference_name="input")

adls_output = PipelineData("output", datastore=adls_datastore)
notebook_path=os.getenv("DATABRICKS_NOTEBOOK_PATH", "/Shared/test") # Databricks notebook path

dbNbStep = DatabricksStep(
    existing_cluster_id="clusterId",
    name="DBNotebookInWS",
    inputs=[adls_input],
    outputs=[adls_output],
    notebook_path=notebook_path,
    notebook_params={'myparam2': pipeline_param},
    run_name='DB_Notebook_demo',
    compute_target=databricks_compute,
    allow_reuse=False
)
````

**Comment by sanpil** [comment]
@fahdkmsft.

**Comment by jarandaf** [comment]
I have also faced the DBFS limitation for existing Databricks clusters when I tried to use output data in a `DatabricksStep`. I am now trying to create a user-defined cluster but I am getting the following error:

`Failed to submit job because of error: Invalid spark_conf parameter`

Isn't `spark_conf` a simple dict with key/value Spark configuration variables? I have not found any example trying to set up a custom Databricks cluster so I am not sure what might be wrong. I would appreciate any hint on that matter.
